Db refactor 1

freesig opened the pull request:
### Summary
This is the first PR in the db refactor series.
- Remove from_agent from places that doesn't make sense once we merge dbs.
- Change `HolochainP2pCell` to HolochailP2pDna`.

There's a few little work arounds just to make this PR work until the next which will remove them.
I will post the next PR once this is merged due to the size and potential for conflicts.

Overall there's a lot of noise from name changes but it's a pretty simple PR.


### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

freesig commented on crates/holochain/src/core/workflow/validation_receipt_workflow.rs:
```diff
@@ -19,12 +18,18 @@ mod tests;
     let keystore = vault.keystore();
     // Who we are.
-    let validator = network.from_agent();
+
+    // Temporary workaround until we remove the need for a
+    // single validator in the next PR.
+    let validator = match vault.kind() {
+        DbKind::Cell(id) => id.agent_pubkey().clone(),
+        _ => unreachable!(),
+    };
```
Things like this will be removed in the follow up PRs

freesig commented on crates/kitsune_p2p/direct/src/v1.rs:
```diff
@@ -860,26 +847,26 @@ async fn handle_call(
-    }
-
-    Ok(())
+    // for (op_hash, op_data) in ops {
+    //     let entry = KdEntrySigned::from_wire(op_data.into_boxed_slice())
+    //         .await
+    //         .map_err(KdError::other)?;
+    //     let op_hash = KdHash::from_kitsune_op_hash(&op_hash);
+    //     if &op_hash != entry.hash() {
+    //         return Err("data did not hash to given hash".into());
+    //     }
+    //     let root = KdHash::from_kitsune_space(&space);
+    //     let to_agent = KdHash::from_kitsune_agent(&to_agent);
+
+    //     kdirect.persist.store_entry(root, to_agent, entry).await?;
+    // }
+
+    // Ok(())
+    todo!()
```
@neonphog I'm not sure how to make this work without a `to_agent`. To agent doesn't really make sense once we are gossiping between nodes. Any idea how I can change this to not need it?

neonphog replied:
direct is lagging behind the holochain updates. It is full sync, and has separate stores per cell even for the same space... At this point I'm starting to wonder if it's worth maintaining. Are we actually getting anything out of it?

If we do want to fix it, for now you would probably need to iterate over all the local agents in that space an apply this operation on them all. It will lead to some duplication of work, because a higher level is already publishing to everyone on that node : )

freesig commented on crates/kitsune_p2p/kitsune_p2p/src/test_util/switchboard/switchboard_evt_handler.rs:
```diff
@@ -129,35 +128,34 @@ impl KitsuneP2pEventHandler for SwitchboardEventHandler {
-                );
-            }
-        })))
+        todo!()
+        // ok_fut(Ok(self.sb.share(|sb| {
+        //     let agent = sb
+        //         .node_for_local_agent_hash_mut(&*to_agent)
+        //         .unwrap()
+        //         .local_agent_by_hash_mut(&*to_agent)
+        //         .unwrap();
+        //     for (hash, op_data) in ops {
+        //         let loc8 = hash.get_loc().as_loc8();
+        //         // TODO: allow setting integration status
+        //         agent.ops.insert(
+        //             loc8,
+        //             AgentOpEntry {
+        //                 is_integrated: true,
+        //             },
+        //         );
+        //     }
+        // })))
```
@maackle This is what I was talking to you about. It doesn't make sense to do things like `.node_for_local_agent_hash_mut(&*to_agent)`. Any idea how we can change this?

neonphog commented on crates/holochain/src/core/ribosome/host_fn/call_remote.rs:
```diff
@@ -15,8 +15,11 @@ pub fn call_remote(
             ..
         } => {
+            let from_agent = super::agent_info::agent_info(_ribosome, call_context.clone(), ())?
+                .agent_latest_pubkey;
```
just curiosity, why is it "latest", does the pubkey iterate somehow in this context?

thedavidmeister replied:
@neonphog yeah there's an "initial" as well

neonphog commented on crates/holochain/src/core/workflow/publish_dht_ops_workflow.rs:
```diff
@@ -219,13 +220,13 @@ mod tests {
 
     /// Call the workflow
-    async fn call_workflow(env: EnvWrite, cell_network: HolochainP2pCell) {
+    async fn call_workflow(env: EnvWrite, cell_network: HolochainP2pDna, author: AgentPubKey) {
```
you had elsewhere removed the `cell_` prefix from the network name, do you want to do that here too?

neonphog commented on crates/kitsune_p2p/kitsune_p2p/src/gossip/simple_bloom/step_4_com_loop_inner.rs:
```diff
@@ -157,54 +157,49 @@ pub(crate) async fn step_4_com_loop_inner_incoming(
-                        if finished && con.peer_cert() == *endpoint.cert() {
-                            i.initiate_tgt = None;
-                        }
+            let futs = bloom.inner.share_mut(move |i, _| {
```
Are we ready to delete simple_bloom yet? If everybody's switched over to the default sharded_gossip, we shouldn't spend cycles maintaining something that isn't needed.

freesig replied:
Yeh sounds like a good idea, maybe in a separate PR?

neonphog reviewed (approved):
couple optional suggestions / questions - looks good

freesig commented:
Unfortunately I made a mistake that is a little hard to fix in this PR but will be fixed in a follow up so even though this is approved I won't merge it but instead pull it into `Db refactor 2`. I have addressed all the suggestions though
